### PR TITLE
rel_output

### PR DESCRIPTION
**Pull request by shlee0605**

### CODE REVIEW

File: 3a/reliable.c [hunk 1]
```diff
@@ -79,22 +83,25 @@ void
...
   pkt->cksum = 0;
   uint16_t new = cksum((void*)pkt, (int) n);
   set_host_bytes(pkt);
 
-  if (pkt->len == ACK_PACKET_SIZE) {
-    DEBUG("it is an ACK packet\n");
-    sw_recv_ack(r, pkt->ackno);
+  if(original == new && pkt->len == n) {
+    if (pkt->len == ACK_PACKET_SIZE) {
+      //DEBUG("it is an ACK packet\n");
```
Comment (chcokr) [inline 3a/reliable.c]: these commented-out DEBUGs can stay?

File: 3a/reliable.c [hunk 2]
```diff
@@ -79,22 +83,25 @@ void
...
   // check if the packet is valid using checksum.
   uint16_t original = pkt->cksum;
   pkt->cksum = 0;
   uint16_t new = cksum((void*)pkt, (int) n);
   set_host_bytes(pkt);
 
-  if (pkt->len == ACK_PACKET_SIZE) {
-    DEBUG("it is an ACK packet\n");
-    sw_recv_ack(r, pkt->ackno);
+  if(original == new && pkt->len == n) {
```
Comment (chcokr) [inline 3a/reliable.c]: Could you explain why `pkt->len == n` is important?

File: 3a/reliable.c (hunk 1 above)
Comment (shlee0605) [inline 3a/reliable.c]: I will revert those DEBUGs.

File: 3a/reliable.c [hunk 3]
```diff
@@ -156,7 +163,19 @@ rel_read (rel_t *s)
...
+  int high = r->sw_receiver->highest_acked_pkt;
+  int i;
+  for(i = 1; i <= high; i++) {
+    if(r->written[i] == 0) {
+      packet_t* pkt = &(r->sw_receiver->sliding_window[i]);
+      conn_output(r->c, pkt->data, pkt->len - HEADER_SIZE); 
+      r->written[i] = 1;
+      //the following lines are for debug purpose
+      packet_t pkt_to_print;
+      set_network_bytes_and_checksum(&pkt_to_print, pkt);
```
Comment (chcokr) [inline 3a/reliable.c]: Can you explain why this line is necessary? Does `print_pkt` expect packets in network byte order?

Comment (shlee0605) [inline 3a/reliable.c]: yes

File: 3a/reliable.c (hunk 2 above)
Comment (shlee0605) [inline 3a/reliable.c]: I forgot to revert this part. For test12, I was trying to see if the packet is dropped correctly when the packet length is recorded as 512 while the length of received packet is 511. Checksum test correctly drops the packet with corrupted length, so we don't need `pkt->len ==n`. I will update this.

File: 3a/reliable.c (hunk 3 above)
Comment (chcokr) [inline 3a/reliable.c]: Alright

**Pull request merged**